### PR TITLE
Add more useful toString method for Request

### PR DESCRIPTION
**Pull request by nartamonov**
Currently `Request`s have useless default toString implementations:

```
> Request(Method.GET, URL(Path("/abc")))
val r1: zhttp.http.Request = zhttp.http.Request$$anon$2@448209ff

> Request.ParameterizedRequest(r1, "PARAMS")
val r2: zhttp.http.Request.ParameterizedRequest[String] = zhttp.http.Request$ParameterizedRequest@26e35afd
```

As opposed to `Response`s:

```
> Response()
val rs: zhttp.http.Response = Response(OK,Headers(Chunk()),Empty,Attribute(None,false,false,None))
```

This PR adds more useful implementation of `Request.toString` that could be helpful for debugging purposes.

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/http/Request.scala [hunk 1]
```diff
@@ -23,6 +23,8 @@ trait Request extends HeaderExtension[Request] { self =>
       override def headers: Headers                   = h
       override def remoteAddress: Option[InetAddress] = self.remoteAddress
       override private[zhttp] def bodyAsByteBuf       = self.bodyAsByteBuf
+      override def toString: String                   =
+        s"Request($method, $url, $headers, $remoteAddress)"
```
Comment (smehta91) [inline zio-http/src/main/scala/zhttp/http/Request.scala]: Can we move this method out on trait `Request` ?

**Review by smehta91 — CHANGES_REQUESTED** [review]

File: zio-http/src/main/scala/zhttp/http/Request.scala (hunk 1 above)
Comment (nartamonov) [reply zio-http/src/main/scala/zhttp/http/Request.scala]: Yep, done. Also added a few tests.

File: zio-http/src/test/scala/zhttp/http/RequestSpec.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package zhttp.http
+
+import zio.test.Assertion._
+import zio.test._
+
+object RequestSpec extends DefaultRunnableSpec {
+  def spec = suite("Request")(
+    suite("toString") {
+      test("should produce string representation of a request") {
+        val r = Request()
```
Comment (smehta91) [inline zio-http/src/test/scala/zhttp/http/RequestSpec.scala]: One small change.
Use ZIO.test.gen to generate values for `url`, `method`, `headers`, and `remoteAddress` and assert exact stringified request.
Refer [this](https://github.com/dream11/zio-http/blob/main/zio-http%2Fsrc%2Ftest%2Fscala%2Fzhttp%2Finternal%2FHttpGen.scala)

**Review by smehta91 — CHANGES_REQUESTED** [review]

File: zio-http/src/test/scala/zhttp/http/RequestSpec.scala (hunk 2 above)
Comment (nartamonov) [reply zio-http/src/test/scala/zhttp/http/RequestSpec.scala]: Done.

**Review by smehta91 — APPROVED** [review]

File: zio-http/src/main/scala/zhttp/http/Request.scala [hunk 3]
```diff
@@ -102,6 +102,9 @@ trait Request extends HeaderExtension[Request] { self =>
    */
   def setUrl(url: URL): Request = self.copy(url = url)
 
+  override def toString: String =
+    s"Request($method, $url, $headers, $remoteAddress)"
+
```
Comment (kaushik-barodiya) [inline zio-http/src/main/scala/zhttp/http/Request.scala:93]: Scala doc would be helpful here :)

**Review by amitksingh1490 — DISMISSED** [review]

**Comment by kaushik143** [comment]
Hi @nartamonov we added another field called `version` also in the `Request` Can you please take a back merge and make the necessary changes in the `toString` method.

**Review by kaushik143 — CHANGES_REQUESTED** [review]

File: zio-http/src/main/scala/zhttp/http/Request.scala [hunk 4]
```diff
@@ -102,6 +102,9 @@ trait Request extends HeaderExtension[Request] { self =>
    */
   def setUrl(url: URL): Request = self.copy(url = url)
 
+  override def toString: String =
+    s"Request($method, $url, $headers, $remoteAddress)"
```
Comment (kaushik-barodiya) [inline zio-http/src/main/scala/zhttp/http/Request.scala]: Small change required, we added version information in the Request. Let's add that too in this method

**Review by kaushik143 — CHANGES_REQUESTED** [review]

File: zio-http/src/main/scala/zhttp/http/Request.scala (hunk 3 above)
Comment (nartamonov) [reply zio-http/src/main/scala/zhttp/http/Request.scala:93]: Done.

File: zio-http/src/main/scala/zhttp/http/Request.scala (hunk 4 above)
Comment (nartamonov) [reply zio-http/src/main/scala/zhttp/http/Request.scala]: Done.

**Review by kaushik143 — APPROVED** [review]

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**